myhdl/_traceSignals.py

```diff
@@ -148,7 +149,14 @@ def _writeVcdSigs(f, hierarchy, tracelists):
                 print("$upscope $end", file=f)
         print("$scope module %s $end" % name, file=f)
         for n, s in sigdict.items():
-            if s._val is None:
+            if isinstance(s, _TristateSignal):
```

Comment:
The original code is mostly agnostic about implementation details in Signal, the new code relies on more such details. examples: 
 - with `isinstance()`, we must import `_TristateSignal` and `_TristateDriver`
 - using `s._orival` in addition to `s._val`

Ugly. But I don't see any simple way around it.